driver/health/event_manager_default.go

```diff
@@ -0,0 +1,64 @@
+package health
+
+import (
+	"context"
+
+	"github.com/ory/x/healthx"
+	"github.com/pkg/errors"
+)
+
+type DefaultHealthEventManager struct {
+	evtChan                chan ReadinessProbeEvent
+	listeners              []ReadinessProbe
+	listenerEventTypeCache map[string]ReadinessProbe
+}
+
+func NewDefaultHealthEventManager(listeners ...ReadinessProbe) (*DefaultHealthEventManager, error) {
+	var listenerEventTypeCache = make(map[string]ReadinessProbe)
+	for _, listener := range listeners {
+		for _, events := range listener.EventTypes() {
+			if _, ok := listenerEventTypeCache[events.ReadinessProbeListenerID()]; ok {
+				return nil, errors.WithStack(ErrEventTypeAlreadyRegistered)
+			}
+			listenerEventTypeCache[events.ReadinessProbeListenerID()] = listener
+		}
+	}
+	return &DefaultHealthEventManager{
+		evtChan:                make(chan ReadinessProbeEvent),
+		listeners:              listeners,
+		listenerEventTypeCache: listenerEventTypeCache,
+	}, nil
+}
+
+func (h *DefaultHealthEventManager) Dispatch(event ReadinessProbeEvent) {
+	if event == nil {
+		return
+	}
+	go func() {
+		h.evtChan <- event
+	}()
+}
+
+func (h *DefaultHealthEventManager) Watch(ctx context.Context) {
+	go func() {
+		for {
+			var evt ReadinessProbeEvent
+			select {
+			case evt = <-h.evtChan:
+			case <-ctx.Done():
+				return
+			}
+			if listener, ok := h.listenerEventTypeCache[evt.ReadinessProbeListenerID()]; ok {
+				listener.EventsReceiver(evt)
+			}
```

Comment:
I think it would be cleaner and easier to understand if done like this:

```suggestion
			select {
			case <-ctx.Done():
				return
			}
			case evt, ok := <-h.evtChan:
				if !ok { // channel closed
					return
				}
				if listener, ok := h.listenerEventTypeCache[evt.ReadinessProbeListenerID()]; ok {
					listener.EventsReceiver(evt)
				}
```